common/unsafe/src/main/java/org/apache/spark/unsafe/memory/UnsafeMemoryAllocator.java

```diff
@@ -36,7 +36,7 @@ public MemoryBlock allocate(long size) throws OutOfMemoryError {
 
   @Override
   public void free(MemoryBlock memory) {
-    assert (memory.obj == null) :
+    assert (memory.obj != null && memory.obj instanceof MemoryAddress) :
```

Comment:
Does the assert string need updating now that the check has changed from `obj == null` to `obj != null`?

Reply:
Most likely! Good spot